presto-main/src/main/java/com/facebook/presto/util/QueryInfoUtils.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.util;
+
+import com.google.common.hash.HashCode;
+import com.google.common.hash.HashFunction;
+
+import java.math.BigInteger;
+
+import static com.google.common.hash.Hashing.sha256;
+import static java.nio.charset.StandardCharsets.UTF_8;
+
+public class QueryInfoUtils
+{
+    private QueryInfoUtils() {}
+
+    public static String convertToQueryHash(String query)
```

Comment:
Rename to computeQueryHash

Reply:
> Let's also add a few tests showing that the hash isn't affected by whitespace

As discussed offline, we go through `sqlFormatter` upstream. Query string should be consistently whitespace formatted.